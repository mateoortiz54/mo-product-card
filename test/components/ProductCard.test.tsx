import React from 'react';

import renderer from 'react-test-renderer'
import {ProductCard} from '../../src/components'
import { product } from '../data/products';

const {act} = renderer;


describe('Pruebas en el ProductImage', () => {
  test('debe de mostrar el componente correctamente con la imagen personalizada', () => {
    
    const wrapper = renderer.create(
        <ProductCard product={product}>
            {
                ()=>(
                    <h1>Hola Mundo</h1>    
                )
            }
        </ProductCard>    
    )
    
    expect(wrapper.toJSON()).toMatchSnapshot();

  });

  test('debe de mostrar el componente correctamente con la imagen personalizada', () => {
    
    const wrapper = renderer.create(
        <ProductCard product={product}>
            {
                ({counter, increaseBy})=>(
                    <>
                        <h1>Hola Mundo</h1>
                        <span>{counter}</span>
                        <button onClick={()=>increaseBy(1)}>+1</button>
                    </>
                )            
            }
        </ProductCard>    

    );

    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();

    act(()=>{
        (tree as any).children[2].props.onClick()
    });

    tree = wrapper.toJSON();

    expect((tree as any).children[1].children[0]).toBe('1');


  });

  
  
  
});