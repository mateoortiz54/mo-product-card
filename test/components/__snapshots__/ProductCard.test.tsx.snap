// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el ProductImage debe de mostrar el componente correctamente con la imagen personalizada 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Hola Mundo
  </h1>
</div>
`;

exports[`Pruebas en el ProductImage debe de mostrar el componente correctamente con la imagen personalizada 2`] = `
<div
  className="productCard undefined"
>
  <h1>
    Hola Mundo
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</div>
`;
